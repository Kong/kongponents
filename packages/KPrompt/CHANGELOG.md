# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 6.14.1 (2022-02-25)


### Bug Fixes

* **kbutton:** danger button hover color ([#506](https://github.com/Kong/kongponents/issues/506)) ([546ed5a](https://github.com/Kong/kongponents/commit/546ed5ae728efadaae3fb3ae0ccb1f785be3da03))





# 6.14.0 (2022-02-23)


### Features

* **kprompt:** add actionPending prop ([#504](https://github.com/Kong/kongponents/issues/504)) ([c59df0b](https://github.com/Kong/kongponents/commit/c59df0b95830ca1fdd0d798d16027dbe008dfeef))





# 6.13.0 (2022-02-22)


### Features

* **kprompt:** add kprompt kongponent ([#503](https://github.com/Kong/kongponents/issues/503)) ([12f0669](https://github.com/Kong/kongponents/commit/12f06699345c76cbc690da3e214f5dbb06aa2485))
