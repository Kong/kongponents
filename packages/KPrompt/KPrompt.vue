<template>
  <KModal
    :is-visible="isVisible"
    :title="displayTitle"
    class="k-prompt"
    @keyup.esc="close"
    @keyup.enter="proceed">
    <template v-slot:header-content>
      <div class="k-prompt-header w-100">
        <div class="k-prompt-header-content d-flex w-100">
          <slot name="header-content">
            <KIcon
              v-if="type === 'warning'"
              icon="warning"
              color="var(--white)"
              secondary-color="var(--yellow-400)"
              size="20"
              class="mr-2" />
            {{ displayTitle }}
          </slot>
          <div class="close-button">
            <KButton
              class="non-visual-button"
              @click="close">
              <KIcon
                icon="close"
                color="var(--grey-600)"
                size="15" />
            </KButton>
          </div>
        </div>
        <hr class="divider">
      </div>
    </template>
    <template v-slot:body-content>
      <div class="k-prompt-body">
        <div class="k-prompt-body-content">
          <slot name="body-content">
            {{ message }}
          </slot>

          <div
            v-if="confirmationText"
            class="k-prompt-confirm-text w-100">
            Type "<span class="bold-600">{{ confirmationText }}</span>" to confirm your action.

            <KInput
              v-model="confirmationInput"
              class="pt-2" />
          </div>
        </div>
        <hr class="divider">
      </div>
    </template>
    <template v-slot:footer-content>
      <div class="k-prompt-action-buttons">
        <slot name="action-buttons">
          <KButton
            appearance="outline"
            class="k-prompt-cancel mr-2"
            @click="close">
            {{ cancelButtonText }}
          </KButton>
          <KButton
            :appearance="type === 'danger' ? 'danger' : 'primary'"
            :disabled="disableProceedButton"
            class="k-prompt-proceed"
            @click="proceed">
            {{ actionButtonText }}
          </KButton>
        </slot>
      </div>
    </template>
  </KModal>
</template>

<script>
import KButton from '@kongponents/kbutton/KButton.vue'
import KIcon from '@kongponents/kicon/KIcon.vue'
import KInput from '@kongponents/kinput/KInput.vue'
import KModal from '@kongponents/kmodal/KModal.vue'

export default {
  name: 'KPrompt',
  components: { KButton, KIcon, KInput, KModal },
  props: {
    title: {
      type: String,
      default: ''
    },
    type: {
      type: String,
      default: 'info',
      validator: (value) => {
        return [
          'info',
          'warning',
          'danger'
        ].includes(value)
      }
    },
    message: {
      type: String,
      default: ''
    },
    actionButtonText: {
      type: String,
      default: 'OK'
    },
    cancelButtonText: {
      type: String,
      default: 'Cancel'
    },
    isVisible: {
      type: Boolean,
      default: false
    },
    /**
     * Use this prop to require a confirmation string be typed correctly
     * before the submit button will be enabled.
     */
    confirmationText: {
      type: String,
      default: ''
    }
  },
  data () {
    return {
      confirmationInput: ''
    }
  },
  computed: {
    displayTitle () {
      if (this.title) {
        if (this.type === 'warning') {
          return 'Warning: ' + this.title
        }

        return this.title
      } else if (this.type === 'info') {
        return 'Information'
      }

      return this.capitalize(this.type)
    },
    disableProceedButton () {
      if (!this.confirmationText.length) {
        return false
      }

      return this.confirmationText !== this.confirmationInput
    }
  },
  mounted () {
    document.addEventListener('keydown', this.handleKeydown)
  },

  beforeDestroy () {
    document.removeEventListener('keydown', this.handleKeydown)
  },
  methods: {
    handleKeydown (e) {
      if (this.isVisible) {
        if (e.keyCode === 27) { // `esc` key
          this.close()
        } else if (e.keyCode === 13) { // `enter` key
          this.proceed()
        }
      }
    },
    capitalize (str) {
      const capitalizeRegEx = /(?:^|[\s-:'"])\w/g

      return str.replace(capitalizeRegEx, (a) => a.toUpperCase())
    },
    close () {
      this.confirmationInput = ''
      this.$emit('canceled')
    },
    proceed () {
      if (this.disableProceedButton) {
        return
      }

      this.confirmationInput = ''
      this.$emit('proceed')
    }
  }
}
</script>

<style lang="scss">
@import '~@kongponents/styles/variables';

.k-prompt {
  --KModalBottomMargin: var(--spacing-md);

  .k-modal-dialog.modal-dialog {
    padding: var(--spacing-lg);
    padding-bottom: var(--spacing-md);

    .close-button {
      margin-left: auto;
    }

    .divider {
      /** subtract parents padding from margin to take full width of modal */
      margin-left: calc(var(--spacing-lg) * -1);
      margin-right: calc(var(--spacing-lg) * -1);
      color: var(--grey-300);
    }

    .k-modal-content {
      .k-modal-header.modal-header {
        width: 100%;
        display: flex;

        .close-button .k-button {
          padding-right: 0;
          padding-top: 0;
        }
      }

      .k-modal-body.modal-body .k-prompt-body {
        font-size: var(--type-md);
        text-align: start;
        color: var(--grey-600);
        line-height: 24px;

        .k-prompt-body-content {
          padding-bottom: var(--spacing-lg);

          .k-prompt-confirm-text {
            margin-top: var(--spacing-lg);

            .k-input {
              width: 100%;
            }
          }
        }
      }

      .k-modal-footer.modal-footer {
        .k-prompt-action-buttons {
          margin-left: auto;
        }
      }
    }
  }
}
</style>
