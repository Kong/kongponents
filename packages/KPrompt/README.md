

# @kongponents/kprompt

[![](https://img.shields.io/npm/v/@kongponents/kprompt.svg?style=flat-square)](https://www.npmjs.com/package/@kongponents/kprompt)

```vue
<KPrompt :description="'hello world'">
  Hello from a slot
</KPrompt>
```
