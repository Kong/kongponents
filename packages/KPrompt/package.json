{
  "name": "@kongponents/kprompt",
  "version": "6.32.15",
  "description": "A dialog that prompts a user to take an action",
  "main": "dist/KPrompt.umd.min.js",
  "componentName": "KPrompt",
  "source": "KPrompt.vue",
  "files": [
    "dist",
    "KPrompt.vue"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Kong/kongponents.git"
  },
  "author": "Kong Inc",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/Kong/kongponents/issues"
  },
  "homepage": "https://github.com/Kong/kongponents#readme",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@kongponents/kbutton": "^6.32.15",
    "@kongponents/kicon": "^6.32.15",
    "@kongponents/kinput": "^6.32.15",
    "@kongponents/kmodal": "^6.32.15",
    "@kongponents/styles": "^6.32.15"
  }
}
