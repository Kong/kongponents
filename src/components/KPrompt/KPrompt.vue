<template>
  <KModal
    v-bind="{ ...sanitizedAttrs, ...modalAttributes }"
    :action-button-appearance="actionButtonAppearance"
    :action-button-disabled="actionButtonDisabledValue"
    :action-button-text="actionButtonText"
    :cancel-button-appearance="cancelButtonAppearance"
    :cancel-button-disabled="cancelButtonDisabled"
    :cancel-button-text="cancelButtonText"
    class="k-prompt"
    :input-autofocus="modalAttributes.inputAutofocus === undefined ? true : modalAttributes.inputAutofocus"
    :title="title || 'Confirm your action'"
    :visible="visible"
    @cancel="$emit('cancel')"
    @proceed="$emit('proceed')"
  >
    <template
      v-if="$slots.title"
      #title
    >
      <slot name="title" />
    </template>
    <template #default>
      <div
        v-if="$slots.default || message"
        class="prompt-content"
      >
        <slot name="default">
          <p class="prompt-message">
            {{ message }}
          </p>
        </slot>
      </div>

      <div
        v-if="confirmationText"
        class="prompt-confirmation-container"
      >
        <p class="prompt-confirmation-text">
          {{ confirmationPromptText[0] }} <span class="confirmation-text">"{{ confirmationText }}"</span>
          {{ confirmationPromptText[1] ? confirmationPromptText[1] : '' }}
        </p>
        <KInput
          v-model="confirmationInput"
          :aria-label="`${confirmationPromptText[0]}&quot;${confirmationText}&quot;${(confirmationPromptText[1] ? confirmationPromptText[1] : '')}`"
          autocapitalize="off"
          autocomplete="off"
          data-testid="confirmation-input"
        />
      </div>
    </template>
  </KModal>
</template>

<script lang="ts" setup>
import type { PropType } from 'vue'
import { computed, ref, useAttrs, watch, nextTick } from 'vue'
import KModal from '@/components/KModal/KModal.vue'
import KInput from '@/components/KInput/KInput.vue'
import type { ButtonAppearance, ModalAttributes } from '@/types'

defineOptions({
  inheritAttrs: false,
})

const props = defineProps({
  visible: {
    type: Boolean,
    default: false,
  },
  title: {
    type: String,
    default: '',
  },
  message: {
    type: String,
    default: '',
  },
  actionButtonText: {
    type: String,
    default: 'Confirm',
  },
  actionButtonAppearance: {
    type: String as PropType<ButtonAppearance>,
    default: 'primary',
  },
  actionButtonDisabled: {
    type: Boolean,
    default: false,
  },
  cancelButtonText: {
    type: String,
    default: 'Cancel',
  },
  cancelButtonAppearance: {
    type: String as PropType<ButtonAppearance>,
    default: 'tertiary',
  },
  cancelButtonDisabled: {
    type: Boolean,
    default: false,
  },
  confirmationText: {
    type: String,
    default: '',
  },
  confirmationPrompt: {
    type: String,
    default: 'Type {confirmationText} to confirm your action.',
  },
  modalAttributes: {
    type: Object as PropType<ModalAttributes>,
    default: () => ({}),
  },
})

const attrs = useAttrs()

defineEmits<{
  (e: 'cancel'): void
  (e: 'proceed'): void
}>()

const sanitizedAttrs = computed(() => {
  const attributes = Object.assign({}, attrs) as Record<string, any>

  // delete attributes that are handled through modalAttributes prop
  delete attributes['tabbable-options']
  delete attributes['max-width']
  delete attributes['max-height']
  delete attributes['close-on-backdrop-click']
  delete attributes['input-autofocus']
  // delete KModal props that are not available in KPrompt
  delete attributes['hide-close-icon']
  delete attributes['hide-cancel-button']
  delete attributes['full-screen']

  return attributes
})

const confirmationInput = ref<string>('')

const actionButtonDisabledValue = computed(() => {
  if (props.actionButtonDisabled) {
    return true
  }

  return props.confirmationText ? props.confirmationText !== confirmationInput.value : false
})

const confirmationPromptText = computed((): string[] => {
  return props.confirmationPrompt.split('{confirmationText}')
})

watch(() => props.visible, async (visible: boolean) => {
  if (visible) {
    await nextTick()
    confirmationInput.value = ''
  }
})
</script>

<style lang="scss" scoped>
.k-prompt {
  .prompt-content + .prompt-confirmation-container {
    margin-top: var(--kui-space-80, $kui-space-80);
  }

  .prompt-confirmation-container {
    display: flex;
    flex-direction: column;
    gap: var(--kui-space-40, $kui-space-40);

    .prompt-confirmation-text {
      .confirmation-text {
        font-weight: var(--kui-font-weight-semibold, $kui-font-weight-semibold);
      }
    }
  }
}
</style>
