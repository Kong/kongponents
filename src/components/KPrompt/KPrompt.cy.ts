import { mount } from 'cypress/vue'
import KPrompt from '@/components/KPrompt/KPrompt.vue'

describe('KPrompt', () => {
  it('renders closed when visible is false', () => {
    mount(KPrompt, {
      props: {
        visible: false,
      },
    })

    cy.get('.k-prompt').should('not.to.exist')
  })

  it('renders open when visible is true', () => {
    mount(KPrompt, {
      props: {
        visible: true,
      },
    })

    cy.get('.k-prompt .modal-container').should('be.visible')
  })

  it('renders action buttons and close icon by default', () => {
    mount(KPrompt, {
      props: {
        visible: true,
      },
    })

    cy.getTestId('modal-close-icon').should('be.visible')
    cy.getTestId('prompt-cancel-button').should('be.visible')
    cy.getTestId('prompt-action-button').should('be.visible')
  })

  it('renders title when passed through prop', () => {
    const title = 'Modal Title'

    mount(KPrompt, {
      props: {
        visible: true,
        title,
      },
    })

    cy.get('.modal-title').should('be.visible').contains(title)
  })

  it('renders title when passed through slot', () => {
    const title = 'Slotted Title'

    mount(KPrompt, {
      props: {
        visible: true,
        title: 'Modal Title',
      },
      slots: {
        title,
      },
    })

    cy.get('.modal-title').should('be.visible').contains(title)
  })

  it('renders message when passed through prop', () => {
    const message = 'Modal Message'

    mount(KPrompt, {
      props: {
        visible: true,
        message,
      },
    })

    cy.get('.prompt-content .prompt-message').should('be.visible').contains(message)
  })

  it('renders content passed through default slot over message prop', () => {
    const content = 'Modal Content'

    mount(KPrompt, {
      props: {
        visible: true,
        message: 'Modal Message',
      },
      slots: {
        default: content,
      },
    })

    cy.get('.prompt-content').should('be.visible').contains(content)
    cy.get('.prompt-content .prompt-message').should('not.exist')
  })

  it('renders action button properly when text, appearance and disabled props are passed', () => {
    const text = 'Action Button'
    const appearance = 'danger'
    const disabled = true

    mount(KPrompt, {
      props: {
        visible: true,
        actionButtonText: text,
        actionButtonAppearance: appearance,
        actionButtonDisabled: disabled,
      },
    })

    cy.getTestId('prompt-action-button').should('be.visible').contains(text).should('have.class', appearance).should('be.disabled')
  })

  it('renders cancel button properly when text, appearance and disabled props are passed', () => {
    const text = 'Cancel Button'
    const appearance = 'danger'
    const disabled = true

    mount(KPrompt, {
      props: {
        visible: true,
        cancelButtonText: text,
        cancelButtonAppearance: appearance,
        cancelButtonDisabled: disabled,
      },
    })

    cy.getTestId('prompt-cancel-button').should('be.visible').contains(text).should('have.class', appearance).should('be.disabled')
  })

  it('renders modal with correct width when passed through modalAttributes prop', () => {
    const widthHeight = '123px'

    mount(KPrompt, {
      props: {
        visible: true,
        modalAttributes: {
          width: widthHeight,
          maxHeight: widthHeight,
        },
      },
    })

    cy.get('.k-prompt .modal-container').should('have.css', 'width', widthHeight)
    cy.get('.k-prompt .modal-content').should('have.css', 'max-height', widthHeight)
  })

  it('emits proceed event when action button is clicked', () => {
    mount(KPrompt, {
      props: {
        visible: true,
      },
    })

    cy.getTestId('prompt-action-button').click().then(() => {
      cy.wrap(Cypress.vueWrapper.emitted()).should('have.property', 'proceed')
    })
  })

  it('emits canceled event when cancel button is clicked', () => {
    mount(KPrompt, {
      props: {
        visible: true,
      },
    })

    cy.getTestId('prompt-cancel-button').click().then(() => {
      cy.wrap(Cypress.vueWrapper.emitted()).should('have.property', 'canceled')
    })
  })

  it('emits canceled event when backdrop is clicked and closeOnBackdropClick is true', () => {
    mount(KPrompt, {
      props: {
        visible: true,
        modalAttributes: {
          closeOnBackdropClick: true,
        },
      },
    })

    cy.get('.k-prompt .modal-backdrop').click('topRight').then(() => {
      cy.wrap(Cypress.vueWrapper.emitted()).should('have.property', 'canceled')
    })
  })

  it('renders confirmation input field and confirmation prompt properly when props are passed', () => {
    const confirmationText = 'Confirmation Text'
    const confirmationPrompt = 'Confirmation {confirmationText} Prompt'

    mount(KPrompt, {
      props: {
        visible: true,
        confirmationPrompt,
        confirmationText,
      },
    })

    cy.get('.prompt-confirmation-text').should('be.visible').contains('Confirmation "Confirmation Text" Prompt')
    cy.getTestId('confirmation-input').should('be.visible')
  })

  it('action button behaves correctly when confirmationText is passed', () => {
    const confirmationText = 'Confirmation Text'

    mount(KPrompt, {
      props: {
        visible: true,
        confirmationText,
        actionButtonDisabled: false,
      },
    })

    cy.getTestId('prompt-action-button').should('be.disabled')

    cy.getTestId('confirmation-input').type(confirmationText).then(() => {
      cy.getTestId('prompt-action-button').should('not.be.disabled')
    })
  })
})
