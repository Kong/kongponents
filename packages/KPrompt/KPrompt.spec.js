import { mount } from '@vue/test-utils'
import KPrompt from '@/KPrompt/KPrompt'

describe('KPrompt', () => {
  it('renders proper content when using slots', () => {
    const headerText = 'This is some header text'
    const bodyText = 'This is some body text'
    const actionButtonsText = 'This is some footer text'

    const wrapper = mount(KPrompt, {
      propsData: {
        isVisible: true,
        title: headerText
      },
      slots: {
        'header-content': `<div>${headerText}</div>`,
        'body-content': `<div>${bodyText}</div>`,
        'action-buttons': `<div>${actionButtonsText}</div>`
      }
    })

    expect(wrapper.find('.k-prompt-header').html()).toEqual(expect.stringContaining(headerText))
    expect(wrapper.find('.k-prompt-body').html()).toEqual(expect.stringContaining(bodyText))
    expect(wrapper.find('.k-prompt-action-buttons').html()).toEqual(expect.stringContaining(actionButtonsText))
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders proper content when using props', () => {
    const title = 'Sweet prop title'
    const message = 'Sweet prop content'

    const wrapper = mount(KPrompt, {
      propsData: {
        isVisible: true,
        title,
        message
      }
    })

    expect(wrapper.find('.k-prompt-header').html()).toEqual(expect.stringContaining(title))
    expect(wrapper.find('.k-prompt-body').html()).toEqual(expect.stringContaining(message))
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders custom button text', () => {
    const actionButtonText = 'Sweet prop action button'
    const cancelButtonText = 'Sweet prop cancel button'

    const wrapper = mount(KPrompt, {
      propsData: {
        isVisible: true,
        actionButtonText,
        cancelButtonText
      }
    })

    const buttons = wrapper.findAll('button')

    // button at 0 is close button in top right corner
    expect(buttons.at(1).text()).toBe(cancelButtonText)
    expect(buttons.at(2).text()).toBe(actionButtonText)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('has a pending state', () => {
    const wrapper = mount(KPrompt, {
      propsData: {
        isVisible: true,
        actionPending: true
      }
    })

    expect(wrapper.find('.k-prompt-proceed .kong-icon-spinner').exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('enables correctly with confirmationText', () => {
    const confirmationText = 'I Agree'

    const wrapper = mount(KPrompt, {
      propsData: {
        isVisible: true,
        confirmationText
      },
      attachToDocument: true
    })

    // disabled
    expect(wrapper.find('.k-prompt-action-buttons .k-prompt-proceed[disabled="disabled"]').exists()).toBe(true)

    const input = wrapper.find('[data-testid="confirmation-input"]')

    // bad input, still disabled
    input.setValue(confirmationText + 'x')
    // enable
    input.setValue(confirmationText)
    expect(wrapper.find('.k-prompt-action-buttons .k-prompt-proceed[disabled="disabled"]').exists()).toBe(false)
    wrapper.find('.k-prompt-action-buttons .k-prompt-proceed').trigger('click')

    expect(wrapper.emitted().proceed).toHaveLength(1)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('proceeds when clicking action button', () => {
    const wrapper = mount(KPrompt, {
      propsData: {
        isVisible: true
      },
      attachToDocument: true
    })

    wrapper.find('.k-prompt-action-buttons .k-prompt-proceed').trigger('click')

    expect(wrapper.emitted().proceed).toHaveLength(1)
  })

  it('proceeds when clicking cancel buttons', () => {
    const wrapper = mount(KPrompt, {
      propsData: {
        isVisible: true
      },
      attachToDocument: true
    })

    wrapper.find('.k-prompt-header .close-button button').trigger('click')

    expect(wrapper.emitted().canceled).toHaveLength(1)

    wrapper.find('.k-prompt-action-buttons .k-prompt-cancel').trigger('click')

    expect(wrapper.emitted().canceled).toHaveLength(2)
  })

  it('tears down event listeners', () => {
    const AEL = jest.fn()
    const REL = jest.fn()

    window.document.addEventListener = AEL
    window.document.removeEventListener = REL

    const wrapper = mount(KPrompt, {
      propsData: {
      },
      attachToDocument: true
    })

    wrapper.destroy()
    expect(AEL).toHaveBeenCalled()
    expect(REL).toHaveBeenCalled()
  })
})
