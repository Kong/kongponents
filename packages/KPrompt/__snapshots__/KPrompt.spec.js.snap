// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KPrompt enables correctly with confirmationText 1`] = `
<div aria-label="Information" role="dialog" aria-modal="true" class="k-modal k-prompt">
  <div class="k-modal-backdrop modal-backdrop">
    <div class="k-modal-dialog modal-dialog">
      <div class="k-modal-content modal-content">
        <div role="heading" aria-level="2" class="k-modal-header modal-header mb-5">
          <div class="k-prompt-header w-100">
            <div class="k-prompt-header-content d-flex align-items-center w-100">
              <!---->
              Information
              <div class="close-button"><button type="button" class="k-button non-visual-button medium rounded outline" aria-label="Close">
                  <!----> <span class="kong-icon kong-icon-close"><svg viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
  <title>Close</title>
  <path d="M16 4 3 17m13 0L3 4" stroke="#A3BBCC" stroke-width="2" stroke-linecap="round"></path>
</svg>
</span>
                  <!----></button></div>
            </div>
            <hr class="divider">
          </div>
        </div>
        <div class="k-modal-body modal-body">
          <div class="k-prompt-body">
            <div class="k-prompt-body-content">

              <div class="k-prompt-confirm-text w-100">
                Type "<span class="bold-600">I Agree</span>" to confirm your action.

                <div class="k-input-wrapper pt-2"><input data-testid="confirmation-input" class="form-control k-input k-input-medium">
                  <!---->
                  <!---->
                </div>
              </div>
            </div>
            <hr class="divider">
          </div>
        </div>
        <div class="k-modal-footer modal-footer d-flex">
          <div class="k-prompt-action-buttons"><button type="button" class="k-button k-prompt-cancel mr-2 medium rounded outline">
              <!---->
              Cancel
              <!----></button> <button type="button" class="k-button k-prompt-proceed medium rounded primary" disabled="disabled">
              <!---->
              <!---->
              OK
              <!----></button></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`KPrompt has a pending state 1`] = `
<div aria-label="Information" role="dialog" aria-modal="true" class="k-modal k-prompt">
  <div class="k-modal-backdrop modal-backdrop">
    <div class="k-modal-dialog modal-dialog">
      <div class="k-modal-content modal-content">
        <div role="heading" aria-level="2" class="k-modal-header modal-header mb-5">
          <div class="k-prompt-header w-100">
            <div class="k-prompt-header-content d-flex align-items-center w-100">
              <!---->
              Information
              <div class="close-button"><button type="button" class="k-button non-visual-button medium rounded outline" aria-label="Close">
                  <!----> <span class="kong-icon kong-icon-close"><svg viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
  <title>Close</title>
  <path d="M16 4 3 17m13 0L3 4" stroke="#A3BBCC" stroke-width="2" stroke-linecap="round"></path>
</svg>
</span>
                  <!----></button></div>
            </div>
            <hr class="divider">
          </div>
        </div>
        <div class="k-modal-body modal-body">
          <div class="k-prompt-body">
            <div class="k-prompt-body-content">

              <!---->
            </div>
            <hr class="divider">
          </div>
        </div>
        <div class="k-modal-footer modal-footer d-flex">
          <div class="k-prompt-action-buttons"><button type="button" class="k-button k-prompt-cancel mr-2 medium rounded outline">
              <!---->
              Cancel
              <!----></button> <button type="button" class="k-button k-prompt-proceed medium rounded primary" disabled="disabled"><span class="kong-icon kong-icon-spinner" viewbox="0 0 16 16"><svg viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
  <title>Loading</title>
  <g>
    <path d="M5 10a5 5 0 1 0 5-5" fill="none" stroke="#A3BBCC" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"></path>
  </g>
</svg>
</span>
              OK
              <!----></button></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`KPrompt renders custom button text 1`] = `
<div aria-label="Information" role="dialog" aria-modal="true" class="k-modal k-prompt">
  <div class="k-modal-backdrop modal-backdrop">
    <div class="k-modal-dialog modal-dialog">
      <div class="k-modal-content modal-content">
        <div role="heading" aria-level="2" class="k-modal-header modal-header mb-5">
          <div class="k-prompt-header w-100">
            <div class="k-prompt-header-content d-flex align-items-center w-100">
              <!---->
              Information
              <div class="close-button"><button type="button" class="k-button non-visual-button medium rounded outline" aria-label="Close">
                  <!----> <span class="kong-icon kong-icon-close"><svg viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
  <title>Close</title>
  <path d="M16 4 3 17m13 0L3 4" stroke="#A3BBCC" stroke-width="2" stroke-linecap="round"></path>
</svg>
</span>
                  <!----></button></div>
            </div>
            <hr class="divider">
          </div>
        </div>
        <div class="k-modal-body modal-body">
          <div class="k-prompt-body">
            <div class="k-prompt-body-content">

              <!---->
            </div>
            <hr class="divider">
          </div>
        </div>
        <div class="k-modal-footer modal-footer d-flex">
          <div class="k-prompt-action-buttons"><button type="button" class="k-button k-prompt-cancel mr-2 medium rounded outline">
              <!---->
              Sweet prop cancel button
              <!----></button> <button type="button" class="k-button k-prompt-proceed medium rounded primary">
              <!---->
              <!---->
              Sweet prop action button
              <!----></button></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`KPrompt renders proper content when using props 1`] = `
<div aria-label="Sweet prop title" role="dialog" aria-modal="true" class="k-modal k-prompt">
  <div class="k-modal-backdrop modal-backdrop">
    <div class="k-modal-dialog modal-dialog">
      <div class="k-modal-content modal-content">
        <div role="heading" aria-level="2" class="k-modal-header modal-header mb-5">
          <div class="k-prompt-header w-100">
            <div class="k-prompt-header-content d-flex align-items-center w-100">
              <!---->
              Sweet prop title
              <div class="close-button"><button type="button" class="k-button non-visual-button medium rounded outline" aria-label="Close">
                  <!----> <span class="kong-icon kong-icon-close"><svg viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
  <title>Close</title>
  <path d="M16 4 3 17m13 0L3 4" stroke="#A3BBCC" stroke-width="2" stroke-linecap="round"></path>
</svg>
</span>
                  <!----></button></div>
            </div>
            <hr class="divider">
          </div>
        </div>
        <div class="k-modal-body modal-body">
          <div class="k-prompt-body">
            <div class="k-prompt-body-content">
              Sweet prop content
              <!---->
            </div>
            <hr class="divider">
          </div>
        </div>
        <div class="k-modal-footer modal-footer d-flex">
          <div class="k-prompt-action-buttons"><button type="button" class="k-button k-prompt-cancel mr-2 medium rounded outline">
              <!---->
              Cancel
              <!----></button> <button type="button" class="k-button k-prompt-proceed medium rounded primary">
              <!---->
              <!---->
              OK
              <!----></button></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`KPrompt renders proper content when using slots 1`] = `
<div aria-label="This is some header text" role="dialog" aria-modal="true" class="k-modal k-prompt">
  <div class="k-modal-backdrop modal-backdrop">
    <div class="k-modal-dialog modal-dialog">
      <div class="k-modal-content modal-content">
        <div role="heading" aria-level="2" class="k-modal-header modal-header mb-5">
          <div class="k-prompt-header w-100">
            <div class="k-prompt-header-content d-flex align-items-center w-100">
              <div>This is some header text</div>
              <div class="close-button"><button type="button" class="k-button non-visual-button medium rounded outline" aria-label="Close">
                  <!----> <span class="kong-icon kong-icon-close"><svg viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
  <title>Close</title>
  <path d="M16 4 3 17m13 0L3 4" stroke="#A3BBCC" stroke-width="2" stroke-linecap="round"></path>
</svg>
</span>
                  <!----></button></div>
            </div>
            <hr class="divider">
          </div>
        </div>
        <div class="k-modal-body modal-body">
          <div class="k-prompt-body">
            <div class="k-prompt-body-content">
              <div>This is some body text</div>
              <!---->
            </div>
            <hr class="divider">
          </div>
        </div>
        <div class="k-modal-footer modal-footer d-flex">
          <div class="k-prompt-action-buttons">
            <div>This is some footer text</div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
